Engine_Revit_UI/Query/Discipline.cs

```diff
@@ -35,15 +35,20 @@ public static partial class Query
         /**** Public Methods                            ****/
         /***************************************************/
 
-        static public Discipline Discipline(this FilterRequest filterRequest, RevitSettings revitSettings)
+        static public Discipline Discipline(this IRequest request, RevitSettings revitSettings)
         {
             Discipline? aDiscipline = null;
 
-            aDiscipline = BH.Engine.Adapters.Revit.Query.Discipline(filterRequest);
+            if (!(request is FilterRequest))
+                return oM.Adapters.Revit.Enums.Discipline.Undefined;
+
+            FilterRequest aFilterRequest = (FilterRequest)request;
```

Comment:
I don't see the point of this. Why defining the method for a generic IRequest if then it works only with FilterRequest?
Is there a point in returning `oM.Adapters.Revit.Enums.Discipline.Undefined`?
I'd prefer this method to be changed to either:
- to accept a new, appropriate implementation of IRequest (preferred option to be done earlier than later, since we're here), OR
- to accept only FilterRequest